### PR TITLE
fix(no-large-snapshots): run on files of any type

### LINKED ISSUE #370: no-large-snapshots should not care about file types
Currently, `no-large-snapshots` checks if the file being linted ends w/ the `.js` file extension, preventing it from linting other files, such as `.tsx`, `.ts`, & `.jsx`.

This file check should be removed. 

https://github.com/jest-community/eslint-plugin-jest/blob/85e4195ecb9373befd42018a8fe4f4bb13cfa629/src/rules/no-large-snapshots.js#L88

### PR DESCRIPTION
**Pull request by graphan**
Adjusts `no-large-snapshots` so that it checks files of any type.

closes #370

### CODE REVIEW

**Comment by G-Rath** [comment]
@graphan this failing danger because you've not provided a description for your PR.

@SimenB I think the codecoverage comment has stopped Danger from posting it's comment for some reason?

**Comment by G-Rath** [comment]
Oh actually danger is failing because of an actual error:

```
Request failed [403]: https://api.github.com/repos/jest-community/eslint-plugin-jest/issues/581/comments
Response: {
  "message": "Resource not accessible by integration",
  "documentation_url": "https://developer.github.com/v3/issues/comments/#create-a-comment"
}
Feedback: undefined
```

File: src/rules/__tests__/no-large-snapshots.test.ts [hunk 1]
```diff
@@ -41,11 +41,6 @@ ruleTester.run('no-large-snapshots', rule, {
         'toThrowErrorMatchingInlineSnapshot',
       ),
     },
-    {
-      // "it should return an empty object for non snapshot files"
```
Comment (SimenB) [inline src/rules/__tests__/no-large-snapshots.test.ts:45]: nove it to invalid to validate it fails for non `.js`?

Comment (graphan) [reply src/rules/__tests__/no-large-snapshots.test.ts:45]: I'm sorry @SimenB, I don't get your question. Could you rephrase it?

Comment (graphan) [reply src/rules/__tests__/no-large-snapshots.test.ts:45]: @SimenB bumping up

Comment (SimenB) [reply src/rules/__tests__/no-large-snapshots.test.ts:45]: There is no test that asserts test files with any extension is tested - instead of removing the test that asserts unknown extensions are _not_ tested, change it to verify it _is_ tested

**Comment by SimenB** [comment]
#637
